run_segmentation.py

```diff
@@ -21,63 +27,58 @@ def run(self):
             segmentation(*self.inputs)
         elif self.method == 'instance_segmentation':
+            log.info(f"running instance segmentation worker on {self.gpuid}")
             instance_segmentation(*self.inputs)
         elif self.method == 'segmentation_validation':
-            segmentation_validation_michael(self.inputs, self.gpuid, 'unfiltered')
+            segmentation_validation_michael(*self.inputs)
 
 
-def main(arguments_):
+def main(method_, raw_dir_, supp_dir_, val_dir_, config_):
+    method = method_
 
-    print("CLI arguments provided")
-    inputs = arguments_.raw
-    outputs = arguments_.supplementary
-    channels = arguments_.channels
-    assert len(channels) > 0, "At least one channel must be specified"
+    inputs = raw_dir_
+    outputs = supp_dir_
+    gpus = config.inference.gpus
 
-    n_gpu = arguments_.gpus
-    method = arguments_.method
+    assert len(config_.inference.channels) > 0, "At least one channel must be specified"
 
     # segmentation validation requires raw, supp, and validation definitions
     if method == 'segmentation_validation':
-        if arguments_.validation:
-            TARGET = arguments_.validation
-        else:
+        if not val_dir_:
             raise AttributeError("validation directory must be specified when method=segmentation_validation")
-        if not arguments_.supplementary:
+        if not outputs:
             raise AttributeError("supplemntary directory must be specifie dwhen method=segmentation_validation")
 
     # segmentation requires raw (NNProb), and weights to be defined
     elif method == 'segmentation':
-        if arguments_.weights is None:
+        if config_.inference.weights is None:
             raise AttributeError("Weights supp_dir must be specified when method=segmentation")
-        else:
-            TARGET = arguments_.weights
 
     # instance segmentation requires raw (stack, NNprob), supp (to write outputs) to be defined
     elif method == 'instance_segmentation':
         TARGET = ''
-
     else:
-        raise AttributeError(f"method flag {arguments_.method} not implemented")
+        raise AttributeError(f"method flag {method} not implemented")
 
     # all methods all require
-    if arguments_.fov:
-        sites = arguments_.fov
+    if config_.inference.fov:
+        sites = config.inference.fov
     else:
         # get all "XX-SITE_#" identifiers in raw data directory
         img_names = [file for file in os.listdir(inputs) if (file.endswith(".npy")) & ('_NN' not in file)]
         sites = [os.path.splitext(img_name)[0] for img_name in img_names]
         sites = list(set(sites))
 
     segment_sites = [site for site in sites if os.path.exists(os.path.join(inputs, "%s.npy" % site))]
-    sep = np.linspace(0, len(segment_sites), n_gpu + 1).astype(int)
+    sep = np.linspace(0, len(segment_sites), gpus + 1).astype(int)
 
     processes = []
-    for i in range(n_gpu):
+    for i in range(gpus):
```

Comment:
interesting yes.  So it appears this portion of the pipeline is set to run across multiple GPUs and not allow one to specify an individual GPU (`config.inference.gpu_id` is never used).  There are a couple paths forward then:
1. allow only one of `gpus` or `gpu_id` for this section
2. have code to handle / override behavior.  So if `gpu_id` is provided, it will always ue just one gpu.  Or if `gpus` is provided, it will always attempt to use multiple gpus.
Do you have a preference?

Reply:
For the behavior of gpu id I think it would be more convenient to be able to specify the gpu to use because gpu 0 might not always be available and the script would error out in this case

Reply:
sounds good.  I will change it so that the GPU can be a list or an int.  If it's a list, then it will run segmentation across each of the listed GPUs